### PR TITLE
Add app documentation from the ArcGIS for Developers site

### PR DESCRIPTION
**Pull request by mikewilburn**
This PR adds documentation that was previously only available on the ArcGIS for Developers site. It introduces a new directory structure for storing docs as well as any images referenced within.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+});
+```
+
+## Reverse geocoding
+
+The Map App uses the built-in map magnifier to help users fine tune a location on the map for reverse geocoding. The magnifier appears after a long-press on the map view. Once the long-press is released, the map point is reverse geocoded.
+
+On long press                    | Reverse geocode result
+:-------------------------------:|:-------------------------------------:
+<img src="/docs/images/reverse_geocode.png" width="300"  />|<img src="/doc/images/reverse_geocode2.png" width="300"  />
```
Comment (esreli) [inline docs/index.md]: The second image appears broken.

Comment (mikewilburn) [reply docs/index.md]: Nice catch here. I stared at that relative path for entirely too long to figure out what the issue was...

**Comment by mikewilburn** [comment]
Eli, I made the one change you called out in the `index.md` as well one other small one you didn't, but which I should have caught earlier.

> Visit Esri's[ **example app site** ](https://developers.arcgis.com/example-apps/maps-app-android/)to read more about this app and how the Runtime features are implemented.

I also corrected this line above in the `README.md` file and a host of small formatting/grammar fixes since I was in there editing it. Please have another look?

File: README.md [hunk 2]
```diff
@@ -80,9 +88,11 @@ If there are changes made in the Original repository, you can sync the fork to k
 4. ```git merge upstream/master``` to sync your local `master' branch with `upstream/master`. **Note**: Your local changes will be retained and your fork's master branch will be in sync with the upstream repository.
 
 ## Requirements
+
 * [Android Studio](http://developer.android.com/sdk/index.html)
 
 ## Resources
+
 * [The Maps-App](https://github.com/Esri/maps-app-android/blob/master/maps-app/README.md)
```
Comment (esreli) [inline README.md]: It appears this link is broken, likely because we're referencing the master blob.

Comment (mikewilburn) [reply README.md]: Good one. Are you okay with removing that line altogether? Seems silly to link _to_ the document _from_ the document.

**Pull request merged**